Examples/src/main/kotlin/com/mapbox/vision/examples/POIView.kt

```diff
@@ -0,0 +1,167 @@
+) : View(context, attrs) {
+
+    companion object {
+
+        // POI will start to appear at this distance, starting with transparent and appearing gradually
+        private const val DRAW_LABEL_MIN_DISTANCE_METERS = 250
+
+        // POI will start to appear from transparent to non-transparent during this first meters of showing distance
+        private const val DISTANCE_FOR_ALPHA_APPEAR_METERS = 50
+        private const val LABEL_SIZE_METERS = 8
+        private const val LABEL_ABOVE_GROUND_METERS = 4
+    }
+
+    var poiList: List<POI> = arrayListOf()
+
+    private val paint = Paint()
+    private val uiHandler = Handler()
+    private var poiDrawDataList: MutableList<POIDrawData> = mutableListOf()
+
+    data class POIState(val poi: POI, val distanceToVehicle: Int, val worldCoordinate: WorldCoordinate?)
+
+    private data class POIDrawData(
+        val poiBitmap: Bitmap,
+        val poiBitmapRect: Rect,
+        val poiBitmapAlpha: Int
+    )
+
+    fun updatePOIState(newVehicleLocation: GeoLocation) {
+        if (poiList.isEmpty()) {
+            return
+        }
+        val currentVehicleLatLng = LatLng(newVehicleLocation.geoCoordinate.latitude, newVehicleLocation.geoCoordinate.longitude)
+        val poiStateList = calculatePOIStateListRegardingVehicle(currentVehicleLatLng)
+        val poiStateListToShow = filterPOIStateListToShow(poiStateList)
+        if (poiStateListToShow.isEmpty()) {
+            return
+        }
+        val pointsDrawDataList = preparePOIDrawData(poiStateListToShow)
+        uiHandler.post {
+            this.poiDrawDataList = mutableListOf<POIDrawData>().apply {
+                addAll(pointsDrawDataList)
+            }
+            invalidate()
+        }
+    }
+
+    // Calculate POI distance to vehicle and WorldCoordinates regarding the vehicle
+    private fun calculatePOIStateListRegardingVehicle(currentVehicleLatLng: LatLng): List<POIState> {
+        return poiList.map {
+            val latLng = LatLng(it.latitude, it.longitude)
+            val worldCoordinate = VisionReplayManager.geoToWorld(GeoCoordinate(latLng.latitude, latLng.longitude))
+            val distanceToVehicle = latLng.distanceTo(currentVehicleLatLng).toInt()
+            POIState(it, distanceToVehicle, worldCoordinate)
+        }
+    }
+
+    // Show only POI which is close enough and behind the car
+    private fun filterPOIStateListToShow(poiStateList: List<POIState>): List<POIState> {
+        return poiStateList.filter {
+            val x = it.worldCoordinate?.x ?: 0.0
+            // Check if POI is behind vehicle and close enough to start appearing
+            (x > 0) && (it.distanceToVehicle < DRAW_LABEL_MIN_DISTANCE_METERS)
+        }
+    }
+
+    private fun preparePOIDrawData(poiStateList: List<POIState>): List<POIDrawData> {
+        val poiDrawDataList = mutableListOf<POIDrawData>()
+        for (poiState in poiStateList) {
+            val worldCoordinate = poiState.worldCoordinate ?: continue
```

Comment:
Perhaps this check to `filterPOIStateListToShow` method?

Reply:
Moreover you already have that check there: `it.worldCoordinate?.x ?: 0.0 && x > 0`

Reply:
fixed